File: Python Password Generator/generator.py
# Lance Rogan
# This is a simple Python Program that creates a random password based on user input

# import proper packages
import random
import string

# ask for user input and store appropriately
characters = (int)(input("How many characters would you like in your password? "))
specialCharacters = (int)(input("How many special characters would you like in your password? "))
numbers = (int)(input("How many numbers would you like in your password? "))

# initialize a list called password
password = list()

# this function generates and returns a random letter 
def random_letter():
    randomLetter = random.choice(string.ascii_letters)
    return randomLetter

# this function generates and returns a random special character/punctuation symbol
def random_specialCharacter():
    randomSpecialCharacter = random.choice(string.punctuation)
    return randomSpecialCharacter

# this function generates and returns a random number
def random_number():
    # we could also do this: randomNumber = random.randint(0,9)
    randomNumber = random.choice(string.digits)
    return randomNumber


#These loops iterate through the number of characters, special characters, and numbers
# and adds the randomly generated items to the password list
for c in range(characters):
    password += random_letter()

for c in range(specialCharacters):
    password += random_specialCharacter()

for c in range(numbers):
    password += random_number()

#shuffle the list
random.shuffle(password)
print("Great! Your password is: ", end="")
#loop through and print out the user password back to the user
for i in range(len(password)):
    print (password[i], end="")

File: Java Password Generator/src/generator.java
/* Lance Rogan
 * 
 * This is a very simple program that creates a randomly generated password for a user to use
 * for their login or newly created account
 * 
 */

import java.util.ArrayList;
import java.util.Collections;
import java.util.Random;
import java.util.Scanner;

public class generator {

	public static void main(String[] args) {

		// creating a scanner object to take input from user keyboard
		Scanner input = new Scanner(System.in);

		// print statements to gather user data and preferences for their password
		System.out.print("How many characters would you like in your password? ");
		int characters = input.nextInt();

		System.out.print("How many special characters would you like in your password? ");
		int specialCharacters = input.nextInt();

		System.out.print("How many numbers would you like in your password? ");
		int numbers = input.nextInt();

		System.out.print("Great! Here is your password: ");

		// initializing arraylist of characters, char c and random object
		ArrayList<Character> passwordShuffle = new ArrayList<Character>();
		char c;
		Random random = new Random();

		// for each character required do the following
		for (int i = 0; i < characters; i++) {
			// generate a random character a-z (lowercase)
			c = (char) (random.nextInt(26) + 97);

			// if a random number is even AKA divisible by 2, convert character to uppercase
			if (random.nextInt(100) % 2 == 0) {
				passwordShuffle.add(Character.toUpperCase(c));
			} else {
				// otherwise leave it as lowercase
				passwordShuffle.add(c);
			}

		}

		// for each special character required do the following
		for (int i = 0; i < specialCharacters; i++) {
			// generate a random special character
			c = (char) (random.nextInt(6) + 33);
			// add to password
			passwordShuffle.add(c);

		}
		// for each number required do the following
		for (int i = 0; i < numbers; i++) {
			// generate a random number from 0-9
			c = (char) (random.nextInt(10) + 48);
			// add to password
			passwordShuffle.add(c);

		}

		// shuffle the password
		Collections.shuffle(passwordShuffle);

		// print out the password to the user
		for (int i = 0; i < passwordShuffle.size(); i++) {
			System.out.print(passwordShuffle.get(i));
		}

	}

}
